minard/polling.py

```diff
@@ -256,6 +397,7 @@ def channel_information(conn):
 
     return channel_info
 
+
 def relay_status(conn):
```

Comment:
This isn't part of this pull request, but I just realized that this is not guaranteed to work. There is no guarantee that the crate state for all crates will be present in the detector state for a run. For example, if a crate is off it will have no data here, so I think you'll have to use a dictionary and index by crate.

Reply:
For example, currently the summary produces this error:

```
Traceback (most recent call last):
  File "/usr/lib/python2.6/site-packages/Flask-0.10-py2.6.egg/flask/app.py", line 1836, in __call__
    return self.wsgi_app(environ, start_response)
  File "/raid/home/tlatorre/minard/minard/__init__.py", line 34, in __call__
    return self.app(environ, start_response)
  File "/usr/lib/python2.6/site-packages/Flask-0.10-py2.6.egg/flask/app.py", line 1820, in wsgi_app
    response = self.make_response(self.handle_exception(e))
  File "/usr/lib/python2.6/site-packages/Flask-0.10-py2.6.egg/flask/app.py", line 1403, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/lib/python2.6/site-packages/Flask-0.10-py2.6.egg/flask/app.py", line 1817, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/lib/python2.6/site-packages/Flask-0.10-py2.6.egg/flask/app.py", line 1477, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/lib/python2.6/site-packages/Flask-0.10-py2.6.egg/flask/app.py", line 1381, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/lib/python2.6/site-packages/Flask-0.10-py2.6.egg/flask/app.py", line 1475, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/lib/python2.6/site-packages/Flask-0.10-py2.6.egg/flask/app.py", line 1461, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/raid/home/tlatorre/minard/minard/views.py", line 574, in check_rates_summary
    crate_average, crun, brun = polling_summary(run)
  File "/raid/home/tlatorre/minard/minard/polling.py", line 186, in polling_summary
    if not check_hv_status(relays, types, channel_info, crate, slot, channel):
  File "/raid/home/tlatorre/minard/minard/polling.py", line 355, in check_hv_status
    hv_relay_mask = relays[crate][1] << 32 | relays[crate][0]
IndexError: list index out of range
```